CONTRIBUTING.md

```diff
@@ -1,6 +1,132 @@
+
+```sh
+$ git clone https://github.com/Your_Username/tiberious.git
+```
+
+> This makes a local copy of repository in your machine.
+
+Once you have cloned the `tiberious` repository in Github, move to that folder first using change directory command on linux and Mac.
+
+```sh
+# This will change directory to a folder tiberious
+$ cd tiberious
+```
+
+Move to this folder for all other commands.
+
+### 3. Set it up :arrow_up:
+
+Run the following commands to see that *your local copy* has a reference to *your forked remote repository* in Github :octocat:
+
+```sh
+$ git remote -v
+origin  https://github.com/Your_Username/tiberious.git (fetch)
+origin  https://github.com/Your_Username/tiberious.git (push)
+```
+
+Now, lets add a reference to the original [tiberious](https://github.com/JustAnotherOrganization/tiberious) repository using
+
+```sh
+$ git remote add upstream https://github.com/JustAnotherOrganization/tiberious.git
+```
+
+> This adds a new remote named ***upstream***.
+
+See the changes using
+
+```sh
+$ git remote -v
+origin    https://github.com/Your_Username/tiberious.git (fetch)
+origin    https://github.com/Your_Username/tiberious.git (push)
+upstream  https://github.com/JustAnotherOrganization/tiberious.git (fetch)
+upstream  https://github.com/JustAnotherOrganization/tiberious.git (push)
+```
+
+### 4. Sync it :recycle:
+
+Always keep your local copy of repository updated with the original repository.
+Before making any changes and/or in an appropriate interval, run the following commands *carefully* to update your local repository.
+
+```sh
+# Fetch all remote repositories and delete any deleted remote branches
+$ git fetch --all --prune
+
+# Switch to `master` branch
+$ git checkout master
+
+# Reset local `master` branch to match `upstream` repository's `master` branch
+$ git reset --hard upstream/master
+
+# Push changes to your forked `tiberious` repo
+$ git push origin master
+```
+
+### 5. Adding and updating dependencies
+
+Install godep
+```sh
+$ go get github.com/tools/godep
+```
+Use godep to get all of the dependencies
+```sh
+godep restore
+```
+
+### 6. Ready Steady Go... :turtle: :rabbit2:
+
+Once you have completed these steps, you are ready to start contributing by checking our Issues and creating [pull requests](https://github.com/JustAnotherOrganization/tiberious/pulls).
+
+### 7. Create a new branch :bangbang:
+
+Whenever you are going to make contribution. Please create seperate branch using command and keep your `master` branch clean (i.e. synced with remote branch).
+
+```sh
+# It will create a new branch with name Branch_Name and switch to branch Folder_Name
+$ git checkout -b Folder_Name
+```
+
+Create a seperate branch for contibution and try to use same name of branch as of folder.
+
+To switch to desired branch
+
+```sh
+# To switch from one folder to other
+$ git checkout Folder_Name
+```
+
+To add the changes to the branch. Use
+
+```sh
+# To add all files to branch Folder_Name
+$ git add .
```

Comment:
Would it be better to advise people to add just the files they want to in case they have stale/unused files in their directory?